### PR TITLE
feat(hunyuan_dit): run hunyuan_dit on mindspore

### PR DESCRIPTION
**Pull request by Cui-yshoho**
Add Hunyuan-DiT

### CODE REVIEW

File: examples/hunyuan_dit/README.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+python sample_t2i.py --infer-mode fa --prompt "渔舟唱晚" --no-enhance --use-fp16
+
+# Generate an image with other image sizes.
+python sample_t2i.py --infer-mode fa --prompt "渔舟唱晚" --image-size 1280 768 --use-fp16
+```
+
+More example prompts can be found in [example_prompts.txt](example_prompts.txt)
+
+### Using Previous versions
+
```
Comment (CaitinZhao) [inline examples/hunyuan_dit/README.md:123]: 这部分折叠掉

File: examples/hunyuan_dit/README.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
+<!-- ## **HunyuanDiT** -->
+
+# Make Hunyuan-DiT run on MindSpore
+
+> Hunyuan-DiT : A Powerful Multi-Resolution Diffusion Transformer with Fine-Grained Chinese Understanding.
+> We've tried to provide a completely consistent interface and usage with the [Tencent/HunyuanDiT](https://github.com/Tencent/HunyuanDiT).
+
+
+## Dependencies and Installation
+
```
Comment (CaitinZhao) [inline examples/hunyuan_dit/README.md:10]: 加一下mindspore，cann的版本信息

File: examples/hunyuan_dit/hydit/annotator/dwpose/__init__.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+def create_mask(image_width, image_height, bboxs):
+    mask = np.zeros((image_height, image_width), dtype=np.float32)
+    for bbox in bboxs:
+        x1, y1, x2, y2 = map(int, bbox)
+        mask[y1 : y2 + 1, x1 : x2 + 1] = 1.0
+    return mask
+
+
+# threshold = 0.4
```
Comment (CaitinZhao) [inline examples/hunyuan_dit/hydit/annotator/dwpose/__init__.py:78]: 这部分为什么注释掉

File: examples/hunyuan_dit/hydit/modules/adamw_mint.py [hunk 4]
```diff
@@ -0,0 +1,339 @@
+from __future__ import absolute_import
```
Comment (CaitinZhao) [inline examples/hunyuan_dit/hydit/modules/adamw_mint.py:1]: 这个文件放到 `mindone/trainer`下面吧

File: examples/hunyuan_dit/hydit/modules/attn_layers.py [hunk 5]
```diff
@@ -0,0 +1,384 @@
+import math
+from typing import Optional, Tuple, Union
+
+import mindspore as ms
+from mindspore import nn, ops
+
+from .norm_layers import LayerNorm
+
+try:
+    from mindspore.ops.operations.nn_ops import FlashAttentionScore as FlashAttention
```
Comment (CaitinZhao) [inline examples/hunyuan_dit/hydit/modules/attn_layers.py:10]: 能直接用`from mindone.models.modules.flash_attention import FLASH_IS_AVAILABLE, MSFlashAttention`吗

File: examples/hunyuan_dit/hydit/annotator/dwpose/__init__.py (hunk 3 above)
Comment (Cui-yshoho) [reply examples/hunyuan_dit/hydit/annotator/dwpose/__init__.py:78]: 这部分用在train_deepspeed_controlnet.py，train_deepspeed_controlnet.py我们目前还不支持

File: examples/hunyuan_dit/hydit/train.sh [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+grad_accu_steps=1                                                       # gradient accumulation
+warmup_num_steps=0                                                      # warm-up steps
+lr=0.0001                                                               # learning rate
+ckpt_every=9999999                                                      # create a ckpt every a few steps.
+ckpt_latest_every=9999999                                               # create a ckpt named `latest.pt` every a few steps.
+ckpt_every_n_epoch=2                                                    # create a ckpt every a few epochs.
+epochs=8                                                                # total training epochs
+
+
+sh $(dirname "$0")/run_g.sh \
```
Comment (CaitinZhao) [inline examples/hunyuan_dit/hydit/train.sh:17]:
```suggestion
bash $(dirname "$0")/run_g.sh \
```

File: examples/hunyuan_dit/hydit/utils/tools.py [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+            for k, v in resume_ckpt.items():
+                if k in local_state:
+                    v.set_dtype(local_state[k].dtype)
+                else:
+                    pass  # unexpect key keeps origin dtype
+        elif resume_path.endswith(".ckpt"):
+            resume_ckpt = ms.load_checkpoint(resume_path)
+        ms.load_param_into_net(model, resume_ckpt, strict_load=True)
+
+    # # Resume EMA model
```
Comment (CaitinZhao) [inline examples/hunyuan_dit/hydit/utils/tools.py:216]: 这部分为什么注释掉了？

File: examples/hunyuan_dit/README.md (hunk 1 above)
Comment (Cui-yshoho) [reply examples/hunyuan_dit/README.md:128]: ok

File: examples/hunyuan_dit/README.md (hunk 2 above)
Comment (Cui-yshoho) [reply examples/hunyuan_dit/README.md:15]: ok

Comment (Cui-yshoho) [reply examples/hunyuan_dit/README.md:15]: ok

File: examples/hunyuan_dit/hydit/modules/adamw_mint.py (hunk 4 above)
Comment (Cui-yshoho) [reply examples/hunyuan_dit/hydit/modules/adamw_mint.py]: ok

File: examples/hunyuan_dit/hydit/train.sh (hunk 6 above)
Comment (Cui-yshoho) [reply examples/hunyuan_dit/hydit/train.sh:17]: 这部分脚本和hunyuan保持一致了 所以写的是sh

File: examples/hunyuan_dit/hydit/utils/tools.py (hunk 7 above)
Comment (Cui-yshoho) [reply examples/hunyuan_dit/hydit/utils/tools.py:216]: hunyuan_pt训练默认没有用ema，ms_hunyuan目前还不支持ema功能

**Review by CaitinZhao — APPROVED** [review]

File: examples/hunyuan_dit/IndexKits/bin/idk
Comment (SamitHuang) [inline examples/hunyuan_dit/IndexKits/bin/idk:1]: .py?

**Review by SamitHuang — APPROVED** [review]

File: examples/hunyuan_dit/IndexKits/bin/idk
Comment (Cui-yshoho) [reply examples/hunyuan_dit/IndexKits/bin/idk:1]: 数据处理部分基于HunYuan，这部分和他们一致

File: examples/hunyuan_dit/README.md [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+# 3. Quick start
+python sample_controlnet.py --infer-mode fa --no-enhance --load-key distill --infer-steps 50 --control-type canny --prompt "在夜晚的酒店门前，一座古老的中国风格的狮子雕像矗立着，它的眼睛闪烁着光芒，仿佛在守护着这座建筑。背景是夜晚的酒店前，构图方式是特写，平视，居中构图。这张照片呈现了真实摄影风格，蕴含了中国雕塑文化，同时展现了神秘氛围" --condition-image-path controlnet/asset/input/canny.jpg --control-weight 1.0 --use-fp16
+```
+
+
+## Benchmark
+
+### Performance
+
+Experiments are tested on ascend [910*] with mindspore [2.3.1] graph mode
```
Comment (vigo999) [inline examples/hunyuan_dit/README.md:177]: 去掉 []

**Review by vigo999 — APPROVED** [review]

File: examples/hunyuan_dit/README.md (hunk 8 above)
Comment (Cui-yshoho) [reply examples/hunyuan_dit/README.md]: ok

Comment (Cui-yshoho) [reply examples/hunyuan_dit/README.md]: ok
